core/controllers/admin.py

```diff
@@ -127,57 +132,131 @@ def get(self):
     @require_super_admin
     def post(self):
         """Handles POST requests."""
+        def check_and_update_config_role(old_list, new_list, role):
+            """This function is used to update the role in user settings when
+            a role is changed using config domain. If a user is removed from
+            old_list, its role is changed to exploration editor. If a new user
+            is added to new_list, its role is changed to role
+
+            Args:
+                old_list: list(str). Existing list of usernames with given
+                    role.
+                new_list: list(str). Updated list of usernames with given role.
+                role: str. The role to be used for users in these lists.
+            """
+            for username in old_list:
+                if username not in new_list:
+                    user_services.update_user_role(
+                        user_services.get_user_id_from_username(username),
+                        feconf.ROLE_EXPLORATION_EDITOR)
+
+            for username in new_list:
+                if username not in old_list:
+                    user_services.update_user_role(
+                        user_services.get_user_id_from_username(username),
+                        role)
+
         try:
             if self.payload.get('action') == 'reload_exploration':
                 exploration_id = self.payload.get('exploration_id')
                 self._reload_exploration(exploration_id)
+                self.render_json({})
             elif self.payload.get('action') == 'reload_collection':
                 collection_id = self.payload.get('collection_id')
                 self._reload_collection(collection_id)
+                self.render_json({})
             elif self.payload.get('action') == 'clear_search_index':
                 exp_services.clear_search_index()
+                self.render_json({})
             elif self.payload.get('action') == 'save_config_properties':
                 new_config_property_values = self.payload.get(
                     'new_config_property_values')
                 logging.info('[ADMIN] %s saved config property values: %s' %
                              (self.user_id, new_config_property_values))
+                config_properties = (
+                    config_domain.Registry.get_config_property_schemas())
                 for (name, value) in new_config_property_values.iteritems():
                     config_services.set_property(self.user_id, name, value)
+                    # below checks are for maintaining the sync between roles
```

Comment:
Eventually the code is going to be removed but not after the migration. 
This code is there to keep sync till all use cases of config domain lists (admin usernames, moderator usernames etc) are removed (which is going to be done in milestone 3).
So, the code(even some of the actions here) are going to be removed after that.